app/models/suspended_email.rb

```diff
@@ -0,0 +1,14 @@
+class SuspendedEmail < ApplicationRecord
+  belongs_to :email_address, inverse_of: :suspended_emails
+  validates :digested_base_email, presence: true
+
+  def self.generate_email_digest(email)
+    normalized_email = EmailNormalizer.new(email).normalized_email
+    OpenSSL::Digest::SHA256.hexdigest(normalized_email)
+  end
+
+  def self.blocked_email_address(email)
```

Comment:
I'd rename this to `find_with_email` to match our `User.find_with_email`

```suggestion
  def self.find_with_email(email)
```